csaf-retrieval/src/jvmMain/kotlin/io/github/csaf/sbom/retrieval/RetrievedProvider.kt

```diff
@@ -438,5 +432,52 @@ class RetrievedProvider(val json: Provider) : Validatable {
                     )
                 }
         }
+
+        /**
+         * Asynchronously constructs a [RetrievedProvider] from a JSON found at the specified URL.
+         * The result is wrapped into a [CompletableFuture].
+         *
+         * @param url The URL of the resource to fetch.
+         * @param loader An instance of [CsafLoader] used to retrieve the provider. Defaults to
+         *   [lazyLoader].
+         * @return A future that resolves to a [RetrievedProvider] or an error if the operation
+         *   fails.
+         */
+        @JvmStatic
+        @JvmOverloads
+        fun fromUrlAsync(url: String, loader: CsafLoader = lazyLoader) =
+            ioScope.future { fromUrl(url, loader).getOrThrow() }
+
+        /**
+         * Constructs a [RetrievedProvider] from a JSON found at the specified URL.
+         *
+         * @param url The URL of the provider JSON to fetch.
+         * @param loader An instance of [CsafLoader] used to retrieve the provider. Defaults to
+         *   [lazyLoader].
+         * @return A [Result] containing the retrieved [RetrievedProvider] or an error if the
+         *   operation fails.
+         */
+        suspend fun fromUrl(url: String, loader: CsafLoader = lazyLoader) =
```

Comment:
We can merge this with fromUrlWithContext by just having `RetrievalContext()` as default parameter for `cdx`

Reply:
let's keep it this way